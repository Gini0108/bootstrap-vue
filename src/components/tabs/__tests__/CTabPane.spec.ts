import { mount } from '@vue/test-utils'
import { CTabPane as Component } from '../../../index'

const ComponentName = 'CTabPane'

const defaultWrapper = mount(Component, {
  propsData: {
      
  },
  slots: {
    default: 'Default slot'
  },
})

const customWrapper = mount(Component, {
  propsData: {
    visible: true
  },
  slots: {
    default: 'Default slot'
  },
})

describe(`Loads and display ${ComponentName} component`, () => {
  it('has a name', () => {
    expect(Component.name).toMatch(ComponentName)
  })
  it('renders correctly', () => {
    expect(defaultWrapper.html()).toMatchSnapshot()
  }) 
  it('contain slots and classes', () => {
    expect(defaultWrapper.text()).toContain('Default slot')
    expect(defaultWrapper.classes('tab-pane')).toBe(true)
    expect(defaultWrapper.classes('fade')).toBe(true)
  })
})

describe(`Customize ${ComponentName} component`, () => {
  it('renders correctly', () => {
    expect(customWrapper.html()).toMatchSnapshot()
  }) 
  it('contain slots and classes', () => {
    expect(customWrapper.text()).toContain('Default slot')
    expect(customWrapper.classes('tab-pane')).toBe(true)
    expect(customWrapper.classes('fade')).toBe(true)
    expect(customWrapper.classes('show')).toBe(true)
    expect(customWrapper.classes('active')).toBe(true)
  })
})