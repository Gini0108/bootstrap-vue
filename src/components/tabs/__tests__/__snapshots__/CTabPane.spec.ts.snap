// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customize CTabPane component renders correctly 1`] = `"<div class=\\"tab-pane fade show active\\">Default slot</div>"`;

exports[`Loads and display CTabPane component renders correctly 1`] = `"<div class=\\"tab-pane fade\\">Default slot</div>"`;
